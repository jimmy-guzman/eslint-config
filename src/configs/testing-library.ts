import * as jestDom from "eslint-plugin-jest-dom";
import testingLibrary from "eslint-plugin-testing-library";

import { GLOB_E2E, GLOB_TESTS } from "../constants";

const testingLibraryConfig = () => {
  return [
    {
      name: "jimmy.codes/testing-library",
      files: GLOB_TESTS,
      plugins: {
        "testing-library": testingLibrary,
        "jest-dom": jestDom,
      },
      rules: {
        ...testingLibrary.configs.react.rules,
        ...jestDom.configs["flat/recommended"].rules,
      },
    },
    {
      name: "jimmy.codes/testing-library/disabled",
      files: GLOB_E2E,
      rules: {
        "testing-library/prefer-screen-queries": "off",
      },
    },
  ];
};

export default testingLibraryConfig;
