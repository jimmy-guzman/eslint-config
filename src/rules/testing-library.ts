import * as jestDom from "eslint-plugin-jest-dom";
import testingLibrary from "eslint-plugin-testing-library";

import { type Rules } from "../types";

export const testingLibraryRules = {
  ...testingLibrary.configs.react.rules,
  ...jestDom.configs["flat/recommended"].rules,
} satisfies Rules;
