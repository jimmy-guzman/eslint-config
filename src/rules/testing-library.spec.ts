import { testingLibraryRules } from "./testing-library";

test("should create testingLibrary rules", () => {
  expect(testingLibraryRules).toMatchSnapshot();
});
