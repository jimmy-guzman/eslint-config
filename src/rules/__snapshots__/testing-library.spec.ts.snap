// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create testingLibrary rules 1`] = `
{
  "jest-dom/prefer-checked": "error",
  "jest-dom/prefer-empty": "error",
  "jest-dom/prefer-enabled-disabled": "error",
  "jest-dom/prefer-focus": "error",
  "jest-dom/prefer-in-document": "error",
  "jest-dom/prefer-required": "error",
  "jest-dom/prefer-to-have-attribute": "error",
  "jest-dom/prefer-to-have-class": "error",
  "jest-dom/prefer-to-have-style": "error",
  "jest-dom/prefer-to-have-text-content": "error",
  "jest-dom/prefer-to-have-value": "error",
  "testing-library/await-async-events": [
    "error",
    {
      "eventModule": "userEvent",
    },
  ],
  "testing-library/await-async-queries": "error",
  "testing-library/await-async-utils": "error",
  "testing-library/no-await-sync-events": [
    "error",
    {
      "eventModules": [
        "fire-event",
      ],
    },
  ],
  "testing-library/no-await-sync-queries": "error",
  "testing-library/no-container": "error",
  "testing-library/no-debugging-utils": "warn",
  "testing-library/no-dom-import": [
    "error",
    "react",
  ],
  "testing-library/no-global-regexp-flag-in-query": "error",
  "testing-library/no-manual-cleanup": "error",
  "testing-library/no-node-access": "error",
  "testing-library/no-promise-in-fire-event": "error",
  "testing-library/no-render-in-lifecycle": "error",
  "testing-library/no-unnecessary-act": "error",
  "testing-library/no-wait-for-multiple-assertions": "error",
  "testing-library/no-wait-for-side-effects": "error",
  "testing-library/no-wait-for-snapshot": "error",
  "testing-library/prefer-find-by": "error",
  "testing-library/prefer-presence-queries": "error",
  "testing-library/prefer-query-by-disappearance": "error",
  "testing-library/prefer-screen-queries": "error",
  "testing-library/render-result-naming-convention": "error",
}
`;
